tiledb/sm/array_schema/attribute.h

```diff
@@ -225,9 +225,13 @@ class Attribute {
    */
   static Attribute deserialize(Deserializer& deserializer, uint32_t version);
 
-  /** Dumps the attribute contents in ASCII form in the selected output. */
+  /** Dumps the attribute contents in ASCII form in the selected output file. */
   void dump(FILE* out) const;
 
+  /** Dumps the attribute contents in ASCII form in the selected output string.
+   */
+  void dump(std::string* out) const;
```

Comment:
Pointer arguments for output is a C idiom. All such function should simply return `string`.

Reply:
Thanks for your comments @eric-hughes-tiledb , they are really in the right direction. I did most of the operator overloads you suggested.

Regarding the `dump` functions added in `Filter` subclasses, I can't find a clean way to remove them. This is basically the problem described here: https://stackoverflow.com/a/4571634

The solutions I see:
1. The existing: Add `operator<<` overload for `std::ostream` for the `Filter` base class that calls virtual function `dump` that the subclasses implement.
2. Add `operator<<` overload for `std::ostream` for the `Filter` base class and all the subclasses. But because there is no way to find the subclass to cast before calling `os << my_filter`, a huge `switch` statement is needed.

I would go with the 1st one, changing all `void dump(std::string* out) const override` to `std::string dump() const override` for the `Filter` subclasses given the chance.

What do you think?